fix(service-worker): avoid storing redundant metadata for hashed assets

gkalpak opened the pull request:
The ServiceWorker needs to keep track of some metadata for unhashed asset resources to know if/when they might need to be revalidated. This applies to resources that do not exist on the filesystem at build-time (and thus can be hashed), such as fonts or images loaded from external sources. For hashed resources, this metadata is irrelevant, because the hash is enough to verify that the content hasn't changed and no revalidation is necessary.

Previously, the ServiceWorker would store such metadata for hashed resources as well, even though it would never be used (thus taking up space unnecessarily).

This commit fixes it by not storing metadata for hased resources, i.e. those that are included in an asset-group's `hashes` array.

mary-poppins commented:
You can preview 7ab218d at https://pr42606-7ab218d.ngbuilds.io/.

IgorMinar commented on packages/service-worker/worker/src/assets.ts:
```diff
@@ -450,7 +450,6 @@ export abstract class AssetGroup {
   protected async maybeUpdate(updateFrom: UpdateSource, req: Request, cache: Cache):
```
the commit message has a small typo: "can be hashed" -> "cannot be hashed".

IgorMinar reviewed (approved):
lgtm! thanks

gkalpak committed:
docs(service-worker): add missing comma in example JSON data

gkalpak committed:
fix(service-worker): avoid storing redundant metadata for hashed assets

The ServiceWorker needs to keep track of some metadata for unhashed
asset resources to know if/when they might need to be revalidated. This
applies to resources that do not exist on the filesystem at build-time
(and thus cannot be hashed), such as fonts or images loaded from
external sources. For hashed resources, this metadata is irrelevant,
because the hash is enough to verify that the content hasn't changed and
no revalidation is necessary.

Previously, the ServiceWorker would store such metadata for hashed
resources as well, even though it would never be used (thus taking up
space unnecessarily).

This commit fixes it by not storing metadata for hashed resources, i.e.
those that are included in an asset-group's `hashes` array.

mary-poppins commented:
You can preview b725c78 at https://pr42606-b725c78.ngbuilds.io/.
